incubator/hnc/api/v1alpha2/hnc_config_conversion.go

```diff
@@ -0,0 +1,190 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	"sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha1"
+)
+
+// ConvertTo converts this hncconfig to the Hub version (v1alpha1).
+func (src *HNCConfiguration) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha1.HNCConfiguration)
+
+	// Spec
+	sspts := src.Spec.Types
+	dspts := []v1alpha1.TypeSynchronizationSpec{}
+	for _, sspt := range sspts {
+		dspt := v1alpha1.TypeSynchronizationSpec{}
+		dspt.APIVersion = sspt.APIVersion
+		dspt.Kind = sspt.Kind
+		switch sspt.Mode {
+		case Propagate:
+			dspt.Mode = v1alpha1.Propagate
+		case Ignore:
+			dspt.Mode = v1alpha1.Ignore
+		case Remove:
+			dspt.Mode = v1alpha1.Remove
+		// The following are the invalid modes in v1alpha2, so they should remain
+		// invalid in v1alpha1. See https://github.com/kubernetes-sigs/multi-tenancy/issues/885
+		case SynchronizationMode(v1alpha1.Propagate):
+			dspt.Mode = v1alpha1.SynchronizationMode(Propagate)
+		case SynchronizationMode(v1alpha1.Ignore):
+			dspt.Mode = v1alpha1.SynchronizationMode(Ignore)
+		case SynchronizationMode(v1alpha1.Remove):
+			dspt.Mode = v1alpha1.SynchronizationMode(Remove)
+		default:
+			dspt.Mode = v1alpha1.SynchronizationMode(sspt.Mode)
```

Comment:
This seems a bit odd to me? If someone writes `propagate` in v1a2, it will turn into `Propagate` in v1a1, correct?

I feel like I'd rather see something like `UnknownV1Alpha2Value-propagate` which should make it a lot more clear what's going on?

Reply:
Agreed, will update.